src/app/georeferencer/qgsvectorwarper.cpp

```diff
@@ -0,0 +1,96 @@
+/***************************************************************************
+     qgsvectorwarper.cpp
+     --------------------------------------
+    Date                 :
+    Copyright            :
+    Email                :
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include <cmath>
+#include <cstdio>
+
+#include "qgsgcpgeometrytransformer.h"
+#include "qgsgcplist.h"
+#include "qgsgeorefdatapoint.h"
+#include "qgscoordinatereferencesystem.h"
+#include "qgscoordinatetransformcontext.h"
+#include "qgsvectorlayer.h"
+#include "qgsgeometry.h"
+#include "qgsprocessingutils.h"
+#include "qgsprocessingcontext.h"
+
+#include <QFile>
+#include <QProgressDialog>
+
+#include "qgsvectorwarper.h"
+
+QgsVectorWarper::QgsVectorWarper( QgsGcpTransformerInterface::TransformMethod &method, const QgsGCPList &points,
+                                  const QgsCoordinateReferenceSystem &destCrs )
+  : mDestCRS( destCrs )
+{
+  QVector<QgsPointXY> srcPts;
+  QVector<QgsPointXY> destPts;
+  for ( QgsGeorefDataPoint *pt : qgis::as_const( points ) )
+  {
+    srcPts << pt->pixelCoords();
+    destPts << pt->transCoords();
+  }
+  mTransformer = new QgsGcpGeometryTransformer( method, srcPts, destPts );
+}
+
+bool QgsVectorWarper::executeTransformInplace( QgsVectorLayer *layer )
+{
+  if ( !layer )
+    return false;
+  if ( layer->sourceCrs() != mDestCRS )
+    layer->setCrs( mDestCRS );
+
+  QgsFeatureIterator it = layer->getFeatures( QgsFeatureRequest() );
+  QgsFeature f;
+  QgsGeometry transformed;
+  bool ok;
+  while ( it.nextFeature( f ) )
+  {
+    transformed = mTransformer->transform( f.geometry(), ok );
+    if ( ok )
+      f.setGeometry( transformed );
+  }
+  return true;
+}
+
+bool QgsVectorWarper::executeTransform( const QgsVectorLayer *layer, const QString outputName )
```

Comment:
```suggestion
bool QgsVectorWarper::executeTransform( const QgsVectorLayer *layer, const QString outputName, QgsFeedback* feedback )
```

(and add cancelation/progress reporting to the feedback)

Reply:
Done